pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -884,8 +884,24 @@ type KubeSchedulerConfig struct {
 	// of all the side-effects and consequences when changing it.
 	// +optional
 	KubeMaxPDVols *string `json:"kubeMaxPDVols,omitempty" protobuf:"bytes,2,opt,name=kubeMaxPDVols"`
+	// Profile configures the scheduling profile for the cluster.
+	// If not specified, the used profile is "balanced" (provides the default kube-scheduler behavior).
+	// +optional
+	Profile *SchedulingProfile `json:"profile,omitempty" protobuf:"bytes,3,opt,name=profile,casttype=SchedulingProfile"`
```

Comment:
Let's default this field to its default value (`balanced`) in our API?

Reply:
I was wondering about the same when I was doing the changes. Do we have some guidelines when to default and when not (rely the controller picking the "default" when the field is not specified)?

Reply:
I don't think we have written down this guideline very prominently, but I remember that we agreed at some point in time to default new API fields explicitly in the `gardener-apiserver` if they have some implicit default value (we discussed this at some PR from Martin V., I believe, but I don't remember which one it was exactly).

Feel free to make this more prominent in https://github.com/gardener/gardener/blob/master/docs/development/changing-the-api.md#gardener-api, today we only have this:

> If necessary then implement/adapt defaulting logic defined in the versioned APIs (e.g., `pkg/apis/core/v1beta1/defaults*.go`).